composer.json

```diff
@@ -29,10 +29,10 @@
         }
     ],
     "require": {
-        "php": "^8.1",
-        "cycle/annotated": "^3.5",
+        "php": "8.1 - 8.4",
+        "cycle/annotated": "^4.2",
         "cycle/database": "^2.8",
-        "cycle/migrations": "^3.1.1|^4.0",
+        "cycle/migrations": "^4.0",
```

Comment:
Switching to v4 might need code changes around tokenizer integration. Have you researched this issue?

Reply:
looked at the tests and wrapped the ClassLocator class with the TokenizerEmbeddingLocator and TokenizerEntityLocator classes, that's all for now.

Reply:
If a dependency does not limit us, I would keep the support range as wide as possible.
There are still many Cycle users who use PHP 8.0.
Some users are gradually migrate projects to Yii 3 and Cycle packages. They asked to bring back compatibility with PSR v2 where we removed it.

Reply:
It is not clear how to register listeners or what events to assign them to read entities.